packages/eas-cli/src/graphql/queries/BuildQuery.ts

```diff
@@ -0,0 +1,75 @@
+import gql from 'graphql-tag';
+
+import { graphqlClient, withErrorHandlingAsync } from '../client';
+import { Build } from '../types/Build';
+
+type Filters = Partial<Pick<Build, 'platform' | 'status'>> & {
+  order?: number;
+  limit?: number;
+};
+
+type ArtifactFragmentType = Pick<Build, 'artifacts'>;
+type PlatformAndArtifactFragmentType = Pick<Build, 'platform' | 'artifacts'>;
+
+export class BuildQuery {
+  static async forArtifactByIdAsync(buildId: string): Promise<PlatformAndArtifactFragmentType> {
+    const data = await withErrorHandlingAsync(
+      graphqlClient
+        .query<{ builds: { byId: PlatformAndArtifactFragmentType } }>(
+          gql`
+          {
+            builds {
+              byId(buildId: "${buildId}") {
+                platform,
+                artifacts {
+                  buildUrl
+                }
+              }
+            }
+          }`
+        )
+        .toPromise()
+    );
+
+    return data.builds.byId;
+  }
+
+  static async allArtifactsForAppAsync(
+    appId: string,
+    filters?: Filters
+  ): Promise<ArtifactFragmentType[]> {
+    const filterData = [`appId: "${appId}"`];
+
+    if (filters?.limit) {
+      filterData.push(`limit: ${filters.limit}`);
+    }
+    if (filters?.order) {
+      filterData.push(`order: ${filters.order}`);
+    }
+    if (filters?.platform) {
+      filterData.push(`platform: ${filters.platform}`);
+    }
+    if (filters?.status) {
+      filterData.push(`status: ${filters.status}`);
+    }
+
+    const data = await withErrorHandlingAsync(
+      graphqlClient
+        .query<{ builds: { allForApp: ArtifactFragmentType[] } }>(
+          gql`
+          {
+            builds {
+              allForApp(${filterData.join(', ')}) {
+                artifacts {
+                  buildUrl
+                }
+              }
+            }
+          }`
```

Comment:
```suggestion
          query ($appId: String!, $limit: Int, $offset: Int, $platform: AppPlatform, $status: BuildStatus) {
            builds {
              allForApp(appId: $appId, limit: $limit, offset: $offset, platform: $platform, status: $status) {
                artifacts {
                  buildUrl
                }
              }
            }
          }, { ...filters, appId }
```
Variables should be used here as well. A nice thing about this is that it eliminates the need to generate GraphQL from string snippets.